### ISSUE TITLE
0.5.0 release

### ISSUE THREAD

**Issue by Bertram25** [OpenDungeons/OpenDungeons#418]
This issue is about maintaining a summary of what is to be done/is done for the 0.5.0 release, and for general dev discussion. :)
Everyone feel free to comment as needed.

**General**
- [x] The code is following the general code guidelines/conventions. #91 
- [x] Remove the Culling Manager #170 
- [x] Random crash #200 
- [x] Bugfix replays #427 #428 
- [x] Copyright update to 2015 #425 
- [x] Update the game version #424 
- [x] Backup the last three game logs #567 
- [x] Fix last packaging/installation issues. Done: #356 #366 #400 #202 (on Windows)

**Game play**
- [x] Implement the Fog of War #295 
- [x] Implement Pay day #431 
- [x] Once the fog of war is done, finish the trap vision logic #408 
- [x] Gold dropped on the floor should follow the same mesh logic than the one in treasury rooms. #279 
- [x] Rooms and traps prices should be configurable. #346 
- [x] Implement Spells support and make workers spawned only through the first of them #16
- [x] Improve the game perspective #345 
- [x] Readd default viewpoint support and make the V key cycle the viewpoints #494 
- [x] Implement the library Research tree logic: #411
- [x] AI doesn't repair damaged rooms. #375 
- [x] Picked-up and then dropped gold mesh stays in hand. #549 
- [x] Revise the creature spawning logic #568 
- [x] Update the help window content #548 

**Creatures**
- [x] Workers: Make gold dug on gold tiles carried like the other items. #407 
- [x] Improve Creature vision processing #396 
- [x] Add support to make creature prefer certain jobs. #354 
- [x] Implement creature mood #11 
- [x] Traps are crafted only if needed #469 
- [x] Improve wall claiming order #282 (ground tile claiming was taken care in 0.4.9)
- [x] Research entities should be brought to the temple center. #576 

**UI**
- [x] Fix tooltips behaviour #337 #397 
- [x] Add a check box, permitting to disable replay creation when quitting the game. #406 
- [x] Improve the UI of Combo-boxes/Check-boxes, ... from generic to OD-like. #419
  - [x] Displayed the number of controlled creatures #562 
- [x] Add the research tree GUI - 0.5.0 part done  #517 

**Graphics**
- [x] Start working on light handling and as a first step #446 #453 #456 
- [x] The Keeper hand is missing some animations #350 
- [x] Troll bed with normals to be fixed and included #503 
- [x] New Goblin + bed #438 #449 (Still lacks weapon bones.)
- [x] Fix material errors -  Done: #529 #570
- [x] Marked tiles weird color #541 
- [x] Certain textures are unused and should be dropped or moved to the media repo. #492 
- [x] Check whether certain materials are unused. #579 
- [x] Fix materials logging errors #606 

**Editor:**
- [x] Fix adding/moving lights #19 #20 #486 
- [x] Fix seeing traps in the map editor. #455 
- [x] Make adding portals/temple possible while editing. #192 

**Documentation**
- [x] Work on making the wiki useful #426 

**Bonus content**
- [x] Console code cleaned up and using CEGUI. #134 #299
- [x] Many fixes to workers and fighters behaviour.
- [x] Advanced rules implemented on spell artifact vision following the FOW + call to war spell implemented.
- [x] A lot more option values are configurable in the config files.
- [x] Lighting system fixed!! within #514 
- [x] Skull flag system to see current creatures health + level + team color #519 
- [x] Tileset configuration and biomes (now known as tileset definable per level) support #14 
- [x] The workshop replaced the forge for traps. The forge will be used later to craft creature's equipment. #459 
- [x] Save game #531 #534 
- [x] The keeper hand is used as an icon and is above the GUI #547 #420 
- [x] Chat and event messages handling upgrade. #638 #642 #647 #654 
- [x] Portals are claimable
- [x] Implemented portal waves

**Final balancing**
- [x] Increase research points needed for higher level researches (50 for first row, 75 for second, 90 for third, 100 for fourth as a first try).
- [x] Decrease wages (say by 30%)
- [x] Decrease amount of gold stored in each treasury tile (currently 5000, we can make it 1000)
- [x] Reduce creature walking speed by 10%
- [x] Set armor increase/level to 0.1 for all creatures
- [x] Raise gained experience from battle (+10%)
- [x] Gives some mood points back when hitting a creature ( #649)
- [x] Increase mana cost of Call to War (x2)
- [x] Do something for falling keeper waves
- [x] Reduce workers claiming rate ( #650)
- [x] Increase portal cooldown (+25%)

---
- [x] Have a good lan to test multiplayer + save MP games.
- [x] Review the CREDITS file for potential mistakes.
- [x] And of course, release once all that is done. ;)

**Comment by Danimal696** [comment]
arent we filling this up pretty fast? ;)

**Comment by Bertram25** [comment]
0.5.0 starts to look good indeed. :)

**Comment by akien-mga** [comment]
Ticked a few boxes more. We're getting there! :-)

**Comment by hwoarangmy** [comment]
I will try to make a release ASAP so that @Danimal696 and @eugeneloza can have a look. BTW, is it possible to get access to sourceforge upload ? It will be easier than uploading it to dropbox and needing @akien-mga to download and upload it...

**Comment by akien-mga** [comment]
> BTW, is it possible to get access to sourceforge upload ? It will be easier than uploading it to dropbox and needing @akien-mga to download and upload it...

Ping @Wout-van-Dongen ^^^

**Comment by Bertram25** [comment]
@hwoarangmy Yeah, you should have access, IMHO.

**Comment by Bertram25** [comment]
To do an update in the current development state and as we are ~3/4 of the current milestone, I think we should make a quick overview of the current plans of everyone involved and what you guys think fits to be part of the 0.5.0 milestone, what should be added/removed/changed there.

I'll start:
I'm officially on the research tree and will if nobody does it, on the installation quirks and user level support afterwards. I'm also sorry I can't give OD more time lately, especially with the pace given by @hwoarangmy and @oyvindln (You just rock guys. :D).
As for sounds, and other GUI general improvement (sliders and the options ui), it will have to wait for now unless somebody else takes it.
As for the current goals displayed here, I think they are fine but I'd move the grabbing logic to afterwards the option gui is functional.
How creature equipment is going to be handled is also still a bit away from us until we have the workshop ready, so I guess we should give a bit more time to @Danimal696 to deliver the start of a workshop. :)

**Comment by hwoarangmy** [comment]
IMHO, there are already pretty much improvements compared to the 0.4.9. I think we should focus on deliver a "playable" version of the game even if lacking some cool features (like sounds).
As a player, I would say that the FOW, the research logic in general (gui + ingame logic) and saving game is already enough to justify a new version. And ATM, we have much more (gui improvements, normals, spells, pay day, tilesets, console, ...)
That's why, I would focus on what we are currently doing (ie finish the research gui, savegames, ...) and remove what is not already started (sound, keeper hand missing animations, UI, ...).
That doesn't mean that we shouldn't work on them (if @oyvindln wants to take one of them before we finish reserach gui and savegames, he can do as he wants) but IMHO, they shouldn't be a requirement.

The main reason is that the sooner we release the 0.5.0, the sooner we will have interesting feedback. If we look at SF downloads, this week, there are 24 downloads of the 0.4.9 win32 and only 5 of the last dev snapshot (that probably includes some of our team members).

**Comment by Bertram25** [comment]
(Pinging @oyvindln as he was not in the users notified list.) - EDIT: He was after all, but that can't hurt. ;)

I'll start by the end of it:

> The main reason is that the sooner we release the 0.5.0, the sooner we will have interesting feedback. If we look at SF downloads, this week, there are 24 downloads of the 0.4.9 win32 and only 5 of the last dev snapshot (that probably includes some of our team members).

My thoughts exactly. Seems we're in sync here. ;)

> IMHO, there are already pretty much improvements compared to the 0.4.9. I think we should focus on deliver a "playable" version of the game even if lacking some cool features (like sounds).

Well, this is a sane goal and a way to "secure" a playable version with more recent and major upgrades.

If all the team is ok to focus on releasing a 0.5.0 with the features in place (and finishing the ones started or hide them temporarily) and move the next things to a 0.5.1 milestone (for instance), I'll be more than happy about it. :)
There are then a few things to decide, though:
- Should we create a 0.5 branch now so that:
  - PRO: If the workshop happens, we can merge it to 'development' without merging to 0.5.0 (and decide what we do about it.)
  - PRO: If some new ground-breaking features happen in devel, we can safely avoid adding them in 0.5 (for the sake of avoiding regressions.)
  - ...
  - CON: Managing two branches can be tedious for tests, even if I'll happily handle most of the cherry-picking. :)
  - CON: Commits will have to be carefully done so that cherry-picking to a branch is not too hellish. ;)
  - ...
    If we all agree about going that road, I'll create the necessary branch and milestone. If not, nevermind, but at least we considered it. ;)
    We can also give us a few weeks more and see how it turns in term of features and models.

WDYT everyone?

Best regards,

**Comment by hwoarangmy** [comment]
IMHO, it is not a good idea to have a branch. We don't have that many developpers and I guess we can hold ourselves from making game breaking features...
However, we should decide what we want for the 0.5.0 RC1 ^^

**Comment by Danimal696** [comment]
Im with hwoarangmy, im in favor of releasing our experimental builds on a more regular basis and keep on the same branch

**Comment by Bertram25** [comment]
Ok, so status quo for now (if nobody has something more to say).
Let's give us a few more weeks and we'll see what's done. :)

**Comment by akien-mga** [comment]
I also like the "release early, release often" approach. The current development branch has so many great features, that I wouldn't want newcomers to try OD 0.4.9 but to try the development snapshot directly :-)

So basically, that means we should do the 0.5.0 release soon ;-)

About the branch, it's up to you guys. I also tend to think that there are more cons than pros, since our current workflow ensures that the development branch is more or less a "stable" branch.
Maybe we could use a different approach? Build the 0.5.0 RC in the development branch, and if we want to introduce ground-breaking features, add them in their own branches to be merged to development once 0.5.0 is branched off?

**Comment by akien-mga** [comment]
> Ok, so status quo for now (if nobody has something more to say).
> Let's give us a few more weeks and we'll see what's done. :)

Well I think we're more or less all willing to finalise the current features and move the non-vital stuff to a later milestone, so that we can release 0.5.0 asap.

**Comment by akien-mga** [comment]
BTW we should plan a LAN to test multiplayer games, and maybe also saving and loading multiplayer games if that's already supported.

**Comment by Bertram25** [comment]
Ok, based on all the input, I've been "phasing out" some non-vital issues into the `If time permits` section. Please, have a look there and tell me whether it suits you all.

If everyone is ok, we can move that into the next milestone and focus on what's left for 0.5.0.

**Comment by hwoarangmy** [comment]
There is no difference between single player or multi player concerning saving (even if I never tried multiplayer). A "known" issue being that I've not saved the seat owners in the saved games. That means that when loading a saved game each player can be on any seat (and not on the one he used to be).

Concerning the 0.5.0, I would propose to drop the sound stuff, the UI not started and the keeper hand. We will see what @Danimal696 can bring before we are ready for the 0.5.0.

**Comment by Bertram25** [comment]
Btw, will the next milestone be 0.5.1 or 0.6.0 ? I'd be for 0.6.0 since the level format could evolve and I know some ground-breaking features may happen in the next version. ;)

**Comment by Bertram25** [comment]
> There is no difference between single player or multi player concerning saving (even if I never tried multiplayer). A "known" issue being that I've not saved the seat owners in the saved games. That means that when loading a saved game each player can be on any seat (and not on the one he used to be).

Good to know, but it's not a real issue as there is no way the game can know for sure which player was on which seat, IMHO. Thus, I think you're good on that one. :)

> Concerning the 0.5.0, I would propose to drop the sound stuff, the UI not started and the keeper hand. We will see what @Danimal696 can bring before we are ready for the 0.5.0.

Agreed. I would still ask to see the models and materials error and warnings fixed for 0.5.0 if possible.

**Comment by hwoarangmy** [comment]
> I'd be for 0.6.0 since the level format could evolve and I know some ground-breaking features may happen in the next version. ;)

I agree. At least until we stop working on saving.

**Comment by hwoarangmy** [comment]
> Good to know, but it's not a real issue as there is no way the game can know for sure which player was on which seat, IMHO. Thus, I think you're good on that one.

Well, if we feel like it, we can add a "default nickname" when saving the seat and try to match the players if we find a corresponding one. But IMHO, that's the best we should do. If someone wants to switch seats, I don't see why we should not allow it.

**Comment by akien-mga** [comment]
> Well, if we feel like it, we can add a "default nickname" when saving the seat and try to match the players if we find a corresponding one. But IMHO, that's the best we should do. If someone wants to switch seats, I don't see why we should not allow it.

I agree we don't need to support 1:1 matching for the seats. A player might have changed her nickname, or might even want to continue the LAN from another computer / OS.

I like the idea of saving the nickname from the previous game though, and display it in the lobby. It could help players remember which seat they were playing during their last LAN if they want to keep the same seat (which is likely for most games). The nickname should not be enforced, just provided as an info for the one setting up the server.

**Comment by akien-mga** [comment]
> I'd be for 0.6.0 since the level format could evolve and I know some ground-breaking features may happen in the next version. ;)

Yes, I'm pretty sure that in a couple of months, we'll feel the same way comparing development with 0.5.0 as we do now with 0.5.0 vs 0.4.9. So 0.6.0 will be most fitting for this awesome game that we'll release ;-)

**Comment by Bertram25** [comment]
Ok, let's go for 0.6.0. ;) As for the nickname, I don't think it's worth it but if it's not too much work (as an indication only) and if you're willing to do it, I'm fine with it. :)

**Comment by Bertram25** [comment]
I'll move the non-vital issues there, and we'll move them back if they happen to be done.

**Comment by Bertram25** [comment]
Ok, so all in all, we have left:
- The Research GUI still not done - @Bertram25 <-- Work on this, you £!¨$¨$ (Yeah, this is a self-insult. ;])
- Save support and AI not fixing their damaged rooms - I guess @hwoarangmy 
- Packaging/installation issues - @Bertram25 and @akien-mga 
- Material warnings to fix + obsolete textures to remove (think about the licenses file, please ;]) and adding weapon bones names to the Goblin - @Danimal696 

Afterwards, it's all about cleanups and bugfixes and we can release. :D

**Comment by akien-mga** [comment]
> Have a good lan to test multiplayer + save MP games.

Here's some WIP for this point: http://forum.freegamedev.net/viewtopic.php?f=35&t=6213 :-D

**Comment by akien-mga** [comment]
How about releasing a 0.5.0 beta sometime this week? (Win32 + Linux64 binaries)
We are slowly growing a community on the #opendungeons IRC channel, and most users tend to struggle with the CEGUI dependency when it comes to building the game from source, so if we provide an "official" pre-0.5.0 binary release I think we could see some nice LANs and get some feedback on the current game.

**Comment by Bertram25** [comment]
> so if we provide an "official" pre-0.5.0 binary release I think we could see some nice LANs and get some feedback on the current game.

Isn't it what @hwoarangmy has already done (several times)?

**Comment by akien-mga** [comment]
> > so if we provide an "official" pre-0.5.0 binary release I think we could see some nice LANs and get some feedback on the current game.
> 
> Isn't it what @hwoarangmy has already done (several times)?

It is, but I'd like to do a Linux version of the same snapshot this time, so that we can be sure that Windows and Linux players can do multiplayer games together :-)
So basically, @hwoarangmy should just tell me which commit he used next time he does a snapshot, and I'll provide a Linux binary.

**Comment by hwoarangmy** [comment]
> Isn't it what @hwoarangmy has already done (several times)?

That looks much like it. It least for windows. I will let you guys do it for linux, though.

**Comment by hwoarangmy** [comment]
> So basically, @hwoarangmy should just tell me which commit he used next time he does a snapshot, and I'll provide a Linux binary.

Yes, you are right, I usually use pretty shadock builds usually including the latest sources and some of my commits (because I just use my sources to build it). To make sure we use the same sources, we should decide by the end of the week (ideally thursday) which repo we should build. The easier probably being to make a branch in my repo.

**Comment by Bertram25** [comment]
Btw, I do think what we're lacking now is 5-8 players maps, and some other single player ones. 
It could also be good to see whether we can tweak the current maps to make them more replayable, like permitting to change the seat, the team or the faction in certain of them.

For the release, I plan to upgrade the Bridge map a bit and have a go at some idea I had. ;)

**Comment by hwoarangmy** [comment]
Please do not add maps until savegames works (especially big maps) as it usually requires to change them while patching.

**Comment by Bertram25** [comment]
> Please do not add maps until savegames works (especially big maps) as it usually requires to change them while patching.

Ok, duly noted. Let's add maps for 0.6.0 then.

**Comment by akien-mga** [comment]
I've updated the topic to add a couple new issues for 0.5.0: the spawning logic revision + the small additions I made in #562.

@Bertram25 Since we will probably use this as a basis for the human-readable changelog, maybe the items of the _Bonus content_ section should be classified in the other categories? Or do we want to keep it to differentiate originally planned features from bonus features?

**Comment by akien-mga** [comment]
If we want to use this topic for the changelog, we should probably also name some of the hidden improvements and modifications that @hwoarangmy made while working on the save feature.

**Comment by Bertram25** [comment]
@akien-mga I've been adding bonus content as a sole category since at first it was small but if you think it's getting messy, I trust you upon improving this. :)
I may also have missed a few things. Feel free to add them and thanks a lot for all that. :)

**Comment by hwoarangmy** [comment]
> we should probably also name some of the hidden improvements and modifications that @hwoarangmy made while working on the save feature

Well, it's more bug fixing and code adaptation. No new feature is there besides game saving.

**Comment by Bertram25** [comment]
A "tons of bugfixes and cleanups line" should handle it. :)
We're so close!

**Comment by akien-mga** [comment]
@hwoarangmy, @Bertram25 I think now would be a good time for a development snapshot based on the upstream/development branch. Apart from the marked tiles bug and the packaging stuff, we are basically at the release candidate :)

**Comment by Bertram25** [comment]
> Apart from the marked tiles bug and the packaging stuff, we are basically at the release candidate :)

Yeah, we are. :D

**Comment by hwoarangmy** [comment]
ok. I will build a win32 release tonight based on the upstream/dev branch

**Comment by Bertram25** [comment]
> ok. I will build a win32 release tonight based on the upstream/dev branch

Sorry, now @akien-mga has taken the habit to get his binaries, you'll never get free of him. ;P

**Comment by akien-mga** [comment]
Linux64 portable binary online :-)

**Comment by akien-mga** [comment]
Note to self: check why our data grew by ~15 MB during this release cycle (most likely due to normal textures, but that's worth a check).

**Comment by hwoarangmy** [comment]
@Bertram25 @akien-mga  The linux binary being online, please do not merge anything until tonight so we have same sources for win32 and linux binaries.

**Comment by akien-mga** [comment]
>  The linux binary being online, please do not merge anything until tonight so we have same sources for win32 and linux binaries.

Fine by me :)

**Comment by Bertram25** [comment]
> @Bertram25 @akien-mga The linux binary being online, please do not merge anything until tonight so we have same sources for win32 and linux binaries.

Only doing PR, I promise. :)

**Comment by hwoarangmy** [comment]
> Only doing PR, I promise. :)

I know it's hard for you to avoid pushing the big green button...

**Comment by Bertram25** [comment]
> I know it's hard for you to avoid pushing the big green button...

Where?? Where?

**Comment by alketii** [comment]
PTW

**Comment by Bertram25** [comment]
> PTW

Posting To Watch?
Play To Win?
Professional Time Waster?

**Comment by alketii** [comment]
hehe, all three in that order :dancers: 

I posted so I can get emails when release happens.

**Comment by akien-mga** [comment]
Potential Tournament Winner ;-)

**Comment by hwoarangmy** [comment]
@alketii If you want, you can also have a look at our forum:
http://forum.freegamedev.net/viewforum.php?f=15

In any case, I will post a message tonight once the new dev release is on:
http://forum.freegamedev.net/viewtopic.php?f=45&t=4765&start=500

There won't be much difference between this dev release and the final 0.5.0 (unless we find bugs) ^^

**Comment by hwoarangmy** [comment]
The dev snapshot is released ^^

**Comment by Bertram25** [comment]
Yeah :D

**Comment by alketii** [comment]
Great, why in sourceforge though ?

**Comment by akien-mga** [comment]
> Great, why in sourceforge though ?

Because we've been hosted by SourceForge historically, and they provide a way better bandwidth that we could with a privately owned server :)
Did you have something else in mind?

**Comment by alketii** [comment]
github, since project is hosted here.

**Comment by akien-mga** [comment]
GitHub is good for hosting release tarballs, but not for random files that we would like to upload like dev snapshots or other stuff.
BTW, even GitHub says they deprecated their file hosting feature: https://github.com/OpenDungeons/OpenDungeons/downloads

**Comment by alketii** [comment]
ah, didn't knew that, thanks.

**Comment by akien-mga** [comment]
Guys, this milestone is almost complete :-)
The only things left are linking the icon with the application on Linux (@Bertram25 is on it right now), and cleaning the unused textures/models/materials from the repo.

Sounds like an imminent release :-D

**Comment by hwoarangmy** [comment]
If we want, we can also add the claiming portal patch ^^ The portal wave doesn't sound very hard too, so it can also be added if ready during the week end.
WDYT ?

**Comment by akien-mga** [comment]
> If we want, we can also add the claiming portal patch ^^ The portal wave doesn't sound very hard too, so it can also be added if ready during the week end.

Since you got started and there's no stopping you when in Chuck Norris mode, I'd say go for it :-P

**Comment by Bertram25** [comment]
> If we want, we can also add the claiming portal patch ^^ The portal wave doesn't sound very hard too, so it can also be added if ready during the week end.

Go for it if you think it isn't too ground-breaking a change. :)

**Comment by Bertram25** [comment]
Well, if not, we can always create a branch before merging the PR, anyway.

**Comment by hwoarangmy** [comment]
> Go for it if you think it isn't too ground-breaking a change. :)

No, I don't think it is. Anyway, there is a fix for the editor because I've seen that persistent objects are not removed when the corresponding room is deleted (however, when the map is saved, it works, it's just a visual problem)

**Comment by akien-mga** [comment]
We're ready for the release IMO. How should we proceed? Features keep coming in, it will be hard to freeze development and release the stable version ;-)

**Comment by Danimal696** [comment]
since we all overcome the huge walls that were preventing the very basics, the rest is adding functionalities, its nice to see them rolling :- ) 
I leave it to you to decide when to publish

**Comment by hwoarangmy** [comment]
I will probably not add any new functionality once my last PR is merged. I might try to make a level that uses the portal wave however for the players to be able to test it ^^
IMHO, it will be good for the 0.5.0 RC1 once the materials logging errors will be fixed

**Comment by oyvindln** [comment]
> IMHO, it will be good for the 0.5.0 RC1 once the materials logging errors will be fixed

Sounds good to me. We've sorted most of the issues now. 
I'm only working on making the code more testable at the moment and not adding features.

**Comment by Bertram25** [comment]
Can anyone have a lan tonight or tomorrow night? I can steal one hour or two if it happens.

**Comment by akien-mga** [comment]
I should be around on #opendungeons :)

**Comment by hwoarangmy** [comment]
I will try to join you tonight

**Comment by akien-mga** [comment]
@hwoarangmy @Bertram25 If one of you has the time to cook up a Win32 binary, that might be nice. If you don't it's no big deal either, I'm making a Linux binary for the guys on the IRC channel that might be around, I think most of them are Linux users.

**Comment by Bertram25** [comment]
Ok, see you tonight. I'll be there around 9pm.

**Comment by hwoarangmy** [comment]
Ok. I will try to make a win32 binary ASAP

**Comment by hwoarangmy** [comment]
The binary is online:
http://sourceforge.net/projects/opendungeons/files/Dev%20Snapshots/OpenDungeons-git20150409-Win32.zip/download

**Comment by Bertram25** [comment]
@hwoarangmy My sister is thanking you! ;)

**Comment by hwoarangmy** [comment]
She is welcome ^^

**Comment by akien-mga** [comment]
So basically we're done with this release :-)
Since our last multiplayer game there were a good deal of nice fixes, the only unchecked box above is "Do something about keeper waves". Any idea about this one?

IMO we can make Win32 and Linux snapshots and label them "0.5.0 RC1". I'll upload a Linux binary asap.

**Comment by akien-mga** [comment]
So I have a "OpenDungeons-0.5.0-RC1-Linux64" portable binary at hand now based on the current development branch. I'll wait for your approval about releasing this as RC1 before I upload it though :)

**Comment by hwoarangmy** [comment]
win32 release on its way

**Comment by hwoarangmy** [comment]
win32 release is there:
http://sourceforge.net/projects/opendungeons/files/Dev%20Snapshots/OpenDungeons-0.5.0-RC1-Win32.zip/download

**Comment by Bertram25** [comment]
@hwoarangmy Would you mind if I or somebody else simply remove the strongest waves, so that the game is winnable with affordable chance to become too easy?

**Comment by hwoarangmy** [comment]
No, you can follow your heart

**Comment by Bertram25** [comment]
> No, you can follow your heart

Then, My heart will go on. ;)

**Comment by Bertram25** [comment]
Development has been push to stable, a 0.5 branch has been created and the 0.5.0-RC1 tag has been added :D 
https://github.com/OpenDungeons/OpenDungeons/releases/tag/0.5.0-RC1

We're getting there!

**Comment by hwoarangmy** [comment]
I'm creating the win32 release RC1

@akien-mga I've removed our RC1 since it was not the real RC1 ^^
Could you please rebuild the linux one ?

**Comment by akien-mga** [comment]
>  Could you please rebuild the linux one ?

Sure.

**Comment by Bertram25** [comment]
Yeah :D

**Comment by hwoarangmy** [comment]
The win32 version is here:
http://sourceforge.net/projects/opendungeons/files/Dev%20Snapshots/OpenDungeons-0.5.0-RC1-Win32.zip/download

**Comment by akien-mga** [comment]
Oops I forgot about my build while in a (long, like always) Mageia council meeting. Uploading right now :)

**Comment by hwoarangmy** [comment]
BTW, I've had a try at level falling keeper. Even after using many traps and killing all heroes lurking, the next wave with the 4 level 25 bigknights raped me. Unbeatable as expected ^^

**Comment by Bertram25** [comment]
Fortunately, not anymore. :)

**Comment by Danimal696** [comment]
> level 25 bigknights raped me

We are all joining into the trauma Bertram felt ;), that unit is crying for a new model and name; please dont forget to update webpage as well.
Also whats new in this one since the last build?

**Comment by akien-mga** [comment]
Linux 0.5.0 RC1 is now online too.

**Comment by akien-mga** [comment]
>  Also whats new in this one since the last build?

Just the latest PR from @Bertram25 (#669) which balances the keeper waves a bit and fixes an issue with the chat vs the console.

**Comment by Danimal696** [comment]
uh oh, win32 build has an error, apparently is missing a letter style: LiberationSans-bold.ttf

Edit: its in there, so maybe the linking is wrong?

**Comment by Bertram25** [comment]
Oh my, must be my fault. What does the log say?
Sorry about that.

**Comment by hwoarangmy** [comment]
It says it cannot find LiberationSans-Bold.ttf but it is included in the release. I guess it doesn't search in gui/fonts

**Comment by hwoarangmy** [comment]
It is my fault after all.
I had not seen the file resource.cfg had changed and I had shipped an old one. The new snapshot is currently uploading ^^

**Comment by Bertram25** [comment]
Thanks :D

**Comment by hwoarangmy** [comment]
I will try to be around on IRC tonight to try a lan with the new RC1 release

**Comment by Bertram25** [comment]
Me too. :) Around 9-9:30PM

**Comment by akien-mga** [comment]
I should be around too :)

**Comment by Bertram25** [comment]
Well, we did have lans and fixed a lot of issues. The 0.5.0 version has a quality level far beyond 0.4.9 IMHO. Since we're still in beta, even a rock-solid one, I think we can release now.

If you're ok, I'll close this issue and the milestone once the news are out. :D 
The two other issues are post-partum ones, anyway.

**Comment by akien-mga** [comment]
> Since we're still in beta, even a rock-solid one, I think we can release now.

What about the segfault experienced yesterday? Do we have any clue as to what may have triggered it?

I'll be on #668 now so that we can include it in 0.5.0 final.

**Comment by hwoarangmy** [comment]
No the segfault happened in a win32 release and there is no backtrace :-(
I just hope it is not related to nvidia drivers again...

**Comment by akien-mga** [comment]
So should we release 0.5.0 like that and maybe do a 0.5.1 if/when we identify the cause for the segfault? (+ other backported bug fixes or balance improvements from the development branch)

**Comment by Bertram25** [comment]
IMHO, that would be a good plan. If we have more users testing the feature, we'll have more feedback on potential flaws, especially on Lan/Wan games.
WDYT?

**Comment by hwoarangmy** [comment]
I think we are in beta stage and it is pretty clear that the game may crash. 0.5.0 was the release that changed from a demo to a game. Compared to 0.5.0, I think we should release more often so, yes, IMHO, we can release even if we experienced an unknown segfault.

**Comment by akien-mga** [comment]
OK let's release asap. I'm discussing our plans for the website with @Wout-van-Dongen to get something ready for 0.5.0, even if it's not the definitive version of the website that he has in mind :)

**Comment by Bertram25** [comment]
\O/

**Comment by akien-mga** [comment]
So basically what's left is taking some good-looking screenshots (#678) and preparing the publication of the release announcement on the website. Then we can probably ask the FreeGamer guys to make a post about us :)

**Comment by akien-mga** [comment]
We should also remember to add the release date in RELEASE-NOTES.md before creating the 0.5.0 tag.

**Comment by Bertram25** [comment]
> We should also remember to add the release date in RELEASE-NOTES.md before creating the 0.5.0 tag

Sure, thanks for the all the work done on doc. :)
Quick link to the news preparation topic for handiness:
http://forum.freegamedev.net/viewtopic.php?p=63377#p63377

As for the release date, I've merge howarangmy bt upgrade so we can have better feed-back FTW. :D 
so we can release sometimes really soon like this evening or tomorrow night if you guys wish it.

**Comment by Bertram25** [comment]
From @akien-mga 's PR:

> @Bertram25 Can you merge the current development branch in 0.5?

I've updated both stable and 0.5 branches with both newly merged PR. We're officially ready to release. I'll work on the release news on the website tonight if nobody else does it before me. :D 
Congrats for all the work done.

**Comment by akien-mga** [comment]
IMO we can release this evening if we're sure that @hwoarangmy backtrace upgrade does not introduce regressions in the Win32 build.

**Comment by Bertram25** [comment]
> IMO we can release this evening if we're sure that @hwoarangmy backtrace upgrade does not introduce regressions in the Win32 build.

So far, no regressions as for me. (I played a game to make "sure".) I think the good is far beyond the bad though, as getting bt when players have crashes will help a lot IMHO.

**Comment by Bertram25** [comment]
I'va added the news as a PR with what we have so far.
It's all a matter of tweaking the screenshots and the release date as necessary now.
Feel free to ping me about all that as necessary. :)

**Comment by hwoarangmy** [comment]
> IMO we can release this evening if we're sure that @hwoarangmy backtrace upgrade does not introduce regressions in the Win32 build.

It is just adding a callback when a crash occurs. If there is a bug, it should be when a segfault occurs, so, not a big problem ^^
However, I couldn't find any way so far to have an executable smaller than 45 Mb to have the symbols (needed to display the whole backtrace when the game crashes)

BTW, sorry for not being very active but I'm on holydays right now ^^

**Comment by Bertram25** [comment]
> However, I couldn't find any way so far to have an executable smaller than 45 Mb to have the symbols (needed to display the whole backtrace when the game crashes)

If the starting and running of the exe isn't too hampered by the binary size, I'd be to use debug symbols.
But I let the final decision in your hands.

> BTW, sorry for not being very active but I'm on holydays right now ^^

Np at all. Have fun! Hope you'll sunny ones. :)

**Comment by hwoarangmy** [comment]
> But I let the final decision in your hands.

In reality, it is in yours since I won't be there to build the 0.5.0 release ^^
But I have the same feeling as you do: nowadays, with the current HDD and internet conexions, the binary size is not that relevant...

**Comment by Bertram25** [comment]
> In reality, it is in yours since I won't be there to build the 0.5.0 release ^^

True! @akien-mga  I'll make the binary asap.
@hwoarangmy Yeah, it's holidays time for you. I'll take care of it. ;)

> But I have the same feeling as you do: nowadays, with the current HDD and internet conexions, the binary size is not that relevant...

Ok, I'll see whether I can keep those debug symbols in place.

**Comment by akien-mga** [comment]
@hwoarangmy Have a nice and well-deserved holiday! (at least based on your hard work on OD ;-)).

Should we add a paragraph in the FAQ about how players can retrieve the opendungeons.log file + the Linux stacktrace or the crash.log file and attach it all when making bug reports?

**Comment by hwoarangmy** [comment]
> Should we add a paragraph in the FAQ about how players can retrieve the opendungeons.log file + the Linux stacktrace or the crash.log file and attach it all when making bug reports?

Good idea

**Comment by Bertram25** [comment]
> Should we add a paragraph in the FAQ about how players can retrieve the opendungeons.log file + the Linux stacktrace or the crash.log file and attach it all when making bug reports?

Yes!! :D

**Comment by akien-mga** [comment]
@Bertram25 Do you know how to make the 0.5.0 look as good as 0.4.9 on GitHub?
Cf. the overview page: https://github.com/OpenDungeons/OpenDungeons/releases 0.4.9 is shown as the "Latest release".

And the tag description is just the commit log: https://github.com/OpenDungeons/OpenDungeons/releases/tag/0.5.0
where it was custom for 0.4.9: https://github.com/OpenDungeons/OpenDungeons/releases/tag/0.4.9

**Comment by Bertram25** [comment]
@akien-mga Thanks for the reminder:

This should be fixed now: https://github.com/OpenDungeons/OpenDungeons/releases
The tag had still to be declared as a release.

**Comment by akien-mga** [comment]
> This should be fixed now: https://github.com/OpenDungeons/OpenDungeons/releases

It's perfect now :)

**Comment by akien-mga** [comment]
I have the source tarball and Linux binary tarball ready. I'll using .tar.xz for the source, as it's starting to be widespread enough and we gain some precious MB for mirrors and distros that will host of copy of the source.

For the binary tarball, I'll use .tar.gz though, because Lutris does not handle .tar.xz yet and I don't want to upload another duplicate tarball in Misc/ like I did for 0.4.9 ;)

**Comment by akien-mga** [comment]
@Bertram25 Know what? We'll have to add a fixup commit and redo the 0.5.0 tag ;-)
"make install" is broken on Linux since I changed the names of some doc files.

I think I'll simply skip installing the doc files to the user's system, package managers usually handle those themselves, choosing what's relevant for their package and what's not.

Note that it doesn't affect binaries at all, so if you've done Win32 binaries, they can be released already.

**Comment by akien-mga** [comment]
Ah no I'll keep installing the doc, because I think providing at least the CREDITS file is a must-have, so it's better to enforce it.

**Comment by akien-mga** [comment]
Ok I think I've managed to update the 0.5 branch and recreate the tag to point on the proper commit. I know it's bad to use `git push -f`, but as long as it's only for tags, it shouldn't be too bad :)

**Comment by Bertram25** [comment]
And you did fine. :) Thanks a lot for spotting it. I was about to say the same. Still on the beast but it is coming. ;)

**Comment by akien-mga** [comment]
We have 1 hour left CEST until our release notes are lying about the release date :-p

**Comment by akien-mga** [comment]
Uploading the Linux 64bit binary right now. I'll probably do a 32bit binary tomorrow, but for now you can comment out the line about the Linux 32bit binary when you update the links for 0.5.0.

**Comment by Bertram25** [comment]
> We have 1 hour left CEST until our release notes are lying about the release date :-p

Np, this can be fixed afterwards. ;) I'm not far as well, but it is my first MingW release, so could you try it out once it is uploaded?

**Comment by akien-mga** [comment]
Yep, I can try it on Wine and also boot Win 8.1 to give it a try.

**Comment by akien-mga** [comment]
It looks like @hwoarangmy put some Big Knights in the MinGW-specific code :-P
I've prepared my 32bit chroot again, so the 32bit Linux binary should arrive soon. Afterwards I'm off to bed :-)

**Comment by Bertram25** [comment]
I'm uploading the installer. (Phew...) Sorry for being slow, the newer mingw installation I have, has got a fresh bug and is lacking libiberty, making ld failed with: cannot find -liberty 
I thus disabled the auto-bt for now. Apart from that, it's working fine. I'll eventually fix that and me or @hwoarangmy can later provide a new binary with auto-bt support later on.
Btw, since I have no auto-bt support, I've used the release binary type.

**Comment by Bertram25** [comment]
The file is uploaded the news fixed merged. 
Yes, 0.5.0 is official now :sunglasses: 

I've tried the win32 zip on another computer and it's working fine so far, but feel free to try!
@akien-mga Still ok to ping FreeGameDev?

I'm officially closing this one + the milestone. All what's done is considered post-partum now.

**Comment by hwoarangmy** [comment]
@Bertram25 I've tried the win32 binaries on a non dev computer and it doesn't launch because you have not included 2 needed dlls from MinGW:
- libgcc_s_dw2-1.dll
- libstdc++-6.dll

Apart from that, the game crashes when starting OpenGL renderer but it is a pretty old computer so it might not be able to use that ^^

**Comment by Bertram25** [comment]
On it!! Thanks :D

**Comment by Bertram25** [comment]
New file being uploaded. The game is running fine on my own, but feel free to keep feeding if needed. :)

**Comment by Bertram25** [comment]
It should be ok now. :)

**Comment by hwoarangmy** [comment]
@Bertram25 I've had a look at your release and you seem to have less dlls then I had. That might come from the options you have compiled with. We will see when @Danimal696 will have tried the binaries...

**Comment by akien-mga** [comment]
I've tried the Win32 binary on Linux with wine, and it gives this error:

```
  what():  CEGUI::GenericException in function 'CEGUI::DynamicModule::DynamicModule(const CEGUI::String&)' (C:\devMingW\cegui-0.8.4\cegui\src\DynamicModule.cpp:222) : Failed to load module 'libCEGUIExpatParser.dll': Module not found.
```

So basically libCEGUIExpatParser.dll is missing.

**Comment by akien-mga** [comment]
I compared the provided DLLs with what I bundle for the Linux binary release, and here are the missing ones in the Win32 build. I don't know if all are required though.
- boost_atomic
- boost_date_time
- boost_locale
- boost_unit_test_framework
- bz2
- CEGUICoreWindowRendererSet
- CEGUIExpatParser
- FLAC
- freeimage
- fribidi
- icudata
- icui18n
- ogg
- vorbis
- vorbisenc

There are also some OGRE components that are provided as DLLs in the Win32 build but that I don't seem to link against on Linux:
- OgrePaging
- OgreProperty
- OgreTerrain
- OgreVolume

Anyway, I'm just linking that as additional info, IMO the simplest way to fix the Win32 build is to make sure you bundle at least the same DLLs as in 0.5.0 RC1 (+ zlib).

For future releases, we can probably have a look at building our various libraries with support only of what we use in the game, so that we can drop some useless libraries on Windows or on Linux (for exemple libFLAC on Linux is most likely linked because of sfml-audio, though we don't use FLAC - building sfml-audio without FLAC support might enable us to remove this unneeded dependency).

**Comment by akien-mga** [comment]
Looking at @hwoarangmy's 0.5.0 RC1 build (this one runs fine in Wine), here are the differences:

Missing in 0.5.0 final:
- libboost_locale-mgw48-mt-1_55
- libCEGUICoreWindowRendererSet
- libCEGUIExpatParser
- libiconv-2

Only included in 0.5.0 final:
- libglew
- libglfw
- libjpeg
- liblibpng
- liblua
- libSILLY
- libtoluapp
- libzlib
- mingwm10
- OgrePaging
- OgreProperty
- OgreTerrain
- OgreVolume
- Plugin_OctreeZone

**Comment by Bertram25** [comment]
Sorry for the two times failure.
I'll make the according changes based on previous MinGW releases.
I'll upload it asap. Keep the feed back coming. :)

**Comment by akien-mga** [comment]
Ready to test :-)

For futures releases I'll have a look at how to make minimal builds of our core libraries (Ogre, CEGUI, SFML) to minimise the number of unused bundled dependencies. For CEGUI at least I've already made myself a minimal RPM spec file to build CEGUI only with support for Ogre and the Expat parser, as we use.

**Comment by Danimal696** [comment]
sorry, i missed on the rush, will try now

**Comment by Bertram25** [comment]
Btw, freeimage is statically linked to the Ogre lib, but I don't know about the other deps.
The newest file is being uploaded with deps corresponding to the 0.5.0-rc1 release.
I didn't add zlib since I didn't compile with auto-bt support.

**Comment by Danimal696** [comment]
bad news, win32 fails: 
![error](https://cloud.githubusercontent.com/assets/8299756/7254554/1e84a1d0-e844-11e4-82d8-e3d45f918ae7.jpg)

**Comment by Bertram25** [comment]
Yep, new one still being uploaded. :) I'll ping here once it's done (~15min left)

**Comment by Bertram25** [comment]
It's done!! The new file should be the good one. :D  But fear my noobiness about MingW builds and try it out. ;)
I won't close the issue a third time. Once everyone can start the game, we close this one.

**Comment by akien-mga** [comment]
It works fine in Wine, congrats :-D

**Comment by Danimal696** [comment]
confirmed working in win32

**Comment by Bertram25** [comment]
k, closing finally. ;)
Thanks to everyone for the help on this one. :)
